android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+                border = BorderStroke(constants.borderWidth, color = constants.defaultContentColor),
+                shape = RoundedCornerShape(getBadgeRounding(badgeType)),
+                modifier = modifier.height(getBadgeHeight(badgeType) + (index * 4).dp)
+                    .padding(top = (index * 4).dp)
+            ) {
+                BadgeFiller(
+                    badgeType = badgeType,
+                    vector = vector,
+                    // This is not a mistype, it is a lifehack to make badges same size
+                    text = badge?.text,
+                    isHiddenLayoutFiller = true,
+                    isStackedBadge = true
+                )
+            }
+        }
+        Surface(
+            color = badge?.color ?: Color.Black,
+            contentColor = badge?.textColor ?: constants.defaultContentColor,
+            border = BorderStroke(constants.borderWidth, color = constants.defaultContentColor),
+            shape = RoundedCornerShape(getBadgeRounding(badgeType)),
+            modifier = modifier
+                .height(getBadgeHeight(badgeType) + (altBadgesSublist.size * 4).dp + (constants.borderWidth * 2))
+                .padding(top = (altBadgesSublist.size * 4).dp)
+        ) {
+            BadgeFiller(
+                badgeType = badgeType,
+                vector = vector,
+                text = badge?.text,
+                isHiddenLayoutFiller = false,
+                isStackedBadge = true
+            )
+        }
+        subbadge?.let {
+            Image(
+                subbadge,
+                modifier = Modifier
+                    .width(constants.subBadgeIconWidth)
+                    .height(constants.subBadgeIconHeight)
+                    .padding(end = 3.dp, bottom = 3.dp)
+                    .constrainAs(createRef()) {
+                        bottom.linkTo(parent.bottom)
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+@Composable
+private fun BadgeFiller(
+    badgeType: BadgeType,
+    vector: ImageVector?,
+    text: String?,
+    isHiddenLayoutFiller: Boolean = false,
+    isStackedBadge: Boolean
+) {
+    Row(
+        modifier = Modifier.padding(
+            horizontal = getBadgeHorizontalPadding(badgeType, isStackedBadge),
+            vertical = getBadgeVerticalPadding(badgeType)
+        ),
+        horizontalArrangement = Arrangement.spacedBy(constants.spacer)
+    ) {
+        vector?.let {
+            Image(
+                vector,
+                colorFilter = ColorFilter.tint(AmbientContentColor.current),
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .width(constants.iconWidth)
+                    .height(if (isHiddenLayoutFiller) 0.dp else constants.iconHeight)
```

Comment:
Perhaps `Image` shouldn't be added at all if `isHiddenLayoutFiller == true`? Could replace the whole block with some spacer, e.g. `Box`. Would avoid unnecessarily loading the vector image.